packages/v3-sdk/src/functions/getPoolAddress.ts

```diff
@@ -0,0 +1,31 @@
+import { Token } from '@pancakeswap/swap-sdk-core'
+import invariant from 'tiny-invariant'
+import { Address, Hash } from 'viem'
+import { DEPLOYER_ADDRESSES, FeeAmount } from '../constants'
+import { computePoolAddress } from '../utils'
+
+export type V3PoolOverride = {
+  initCodeHash?: Hash
+  deployerAddress?: Address
+}
+
+/**
+ * returns the address of v3 pool
+ */
+export const getPoolAddress = (tokenA: Token, tokenB: Token, fee: FeeAmount, override?: V3PoolOverride): Address => {
```

Comment:
Let's use `number` instead of `FeeAmount` for utility functions to support dynamic fee tier

Reply:
plus we can add a layer of in memory cache for this function. For example to use `lodash/memoize`

Reply:
I think should remove this file:

- as V4 CLPool, there is not `poolAddress` just `PoolId`
- as V3 Pool, use `computePoolAddress` directly as before